src/main/java/spoon/reflect/declaration/CtTypeInformation.java

```diff
@@ -60,6 +60,11 @@
 	boolean isAnonymous();
 
 	/**
+	 * Return {@code true} if the referenced type is declared in a executable.
+	 */
+	boolean isEnclosingType();
```

Comment:
brainstorm for a better name:

* isInExecutable
* isLocalOrAnonymousClass (as in java.lang.Class)
* isLocalOrAnonymousOrLambdaClass (are all lambdas compiled to classes? )